### DISCUSSION TITLE
-annotate with diacriticals fails

### DISCUSSION THREAD

**Discussion by karel1** [ImageMagick/ImageMagick#6789] [Help] [answered]
### ImageMagick version

7.1.1-20

### Operating system, version and so on

Linux Manjaro kernel 6.1.

### Description

I have a question on font and diacriticals. The short version is that a word like 'Füssen' is shown by ImageMagick as 'F..ssen', the string 'ä ï ë ö ä test ø œ å' is shown as '.. .. .. .. .. test .. .. ..'

I am constructing a batch command to annotate pictures with titles. The mogrify / annotate works fine except for Titles with diacriticals.

#!/usr/bin/env bash
for img in *.jpg;  do 
  exiftool -q -overwrite_original '-Make<Title' $img
  mogrify "$img" -auto-orient -gravity South \\
   -background Black \\
-font Helvetica -pointsize 80 -fill White \\
-splice 0x110 -annotate 0x0  %[exif:Make] $img;
   echo "$img " 
done

Since the tag exif:Title is not recognized by ImageMagick (unknown image property), I am using exiftool to copy the Title into a tag that is recognized. (This is in a copy of the original picture file.)

This is on Linux Manjaro (based on Arch). Diacriticals are fine on this system. Exiftool does show the titles correct.

❯ exiftool -Make test4.jpg
Make                            : ä ï ë ö ä test ø œ å

When %[exif:Make] is replaced by "ä ï ë ö ä test ø œ å" in the mogrify command it completes OK, i.e. the annotation shows up correct.

What am I missing?  Thanks for your help.

**Comment by snibgo** [comment]
As your IM is v7, I suggest using `magick mogrify`, not merely `mogrify`.

Your mogrify command is wrong. It should not start with `"$img"`. See https://imagemagick.org/script/mogrify.php

I suggest you post a sample input so we can try to reproduce the problem.

**Comment by urban-warrior** [comment]
Start with the basics.  Do you get expected diacritical's with this command: `magick -pointsize 80 label:'Füssen' diacritical.png`.

**Comment by karel1** [comment]
Thank you both for your reaction. I have attached 4 files:
a) the result of
magick -pointsize 80 label:'Füssen' output_a.png

![output_a](https://github.com/ImageMagick/ImageMagick/assets/3692937/e3825d85-23d0-4205-a6a0-737d8cd0b029)

b) the result of
magick convert test4.jpg -gravity South -background Black -font Helvetica -pointsize 50 -fill White -splice 0x54 -annotate 0x0 "ü û test ™ ë é" output_b.jpg 

![output_b](https://github.com/ImageMagick/ImageMagick/assets/3692937/7b7b6917-b814-43c7-a755-aea8b293af18)

c) the result of
magick convert test4.jpg -gravity South -background Black -font Helvetica -pointsize 50 -fill White -splice 0x54 -annotate 0x0 "%[exif:Make]" output_c.jpg

![output_c](https://github.com/ImageMagick/ImageMagick/assets/3692937/ea1e6490-f4cc-4eaf-adbc-bd9209ad459c)

and the input file:

![test4](https://github.com/ImageMagick/ImageMagick/assets/3692937/58887275-8ec3-465b-a289-3996fd039e78)

magick test4.jpg -verbose info      shows a.o:
Properties:
    date:create: 2023-10-16T11:43:18+00:00
    date:modify: 2023-10-14T14:19:09+00:00
    date:timestamp: 2023-10-16T12:31:31+00:00
    exif:ExifOffset: 108, 
    exif:Make: .. .. .. .. .. test .. .. ..
    exif:PixelXDimension: 1552, 

so it seems that imagemagick does not read / accept the diacreticals in Make!?

Thanks again!

**Comment by snibgo** [comment]
It seems to work fine for me, IM v7.1.1-20 on Windows 11:
```
magick test4.jpg -pointsize 100 -gravity Center -annotate +0+0 %[EXIF:Make] x.jpg
```

![x](https://github.com/ImageMagick/ImageMagick/assets/33812368/c70d6e03-6397-4c0c-83c0-4d0c599600d5)

**Reply by karel1** [reply]
Interesting! Your test on Windows 11 suggests that the issue is between (my) Linux Manjaro and ImageMagick. I have executed your commandline and the result is still as expected ".. .. .. .. .. test .. .. .." I have tested this on the pure Arch install that is on my multiboot pc as well as a Linux Mint install. The result is the same.

**Comment by snibgo** [comment]
Perhaps your default font doesn't contain glyphs for those characters.

**Reply by karel1** [reply]
The output_a.png shows Füssen as was the objective and the exiftool shows exif:make as it should. Or am I missing your point?

**Comment by snibgo** [comment]
I agree, your tests above suggest it isn't a font problem.

Incidentally, you sometimes use `magick convert` which uses the old v6 syntax, and sometimes just `magick` which uses the newer better v7 syntax. I suggest always using just `magick`.

**Reply by karel1** [reply]
The full output of "magick test4.jpg -verbose info:" shows a.o. also
Orientation: TopLeft
  Profiles:
    Profile-8bim: 102 bytes
    Profile-exif: 158 bytes
    Profile-iptc: 90 bytes
      **Caption[2,120]: ä ï ë ö ä test ø œ å**
      **Headline[2,105]: ä ï ë ö ä test ø œ å**
      Originating Program[2,65]: Shotwell
      Program Version[2,70]: 0.32.2
    Profile-xmp: 2887 bytes
  Properties:
    date:create: 2023-10-16T15:13:10+00:00
    date:modify: 2023-10-16T15:13:10+00:00
    date:timestamp: 2023-10-16T15:13:16+00:00
    exif:ExifOffset: 122, 
    exif:Make: .. .. .. .. .. test .. .. ..

Bold by me. So, there it is, but how can I reference this in -annotate? This does not work: "S %[iptc:Caption[2,120]] F" . There is no error, only "S  F" shows up on the picture.

**Comment by snibgo** [comment, marked as answer]
You can use IPTC like this:
```
magick test4.jpg -pointsize 100 -gravity Center -annotate +0+0 hello%[IPTC:2:120]world x2.jpg
```

**Reply by karel1** [reply]
Great! Problem solved, thank you very much.

**Comment by karel1** [comment]
FYI this is the "final" result, it does the job.

#!/usr/bin/env bash
magick mogrify *.jpg -define preserve-timestamp=true -auto-orient -gravity South \
   -background Black \
-font Helvetica-Bold -pointsize 80 -fill White \
-splice 0x110 -annotate 0x0  "%[IPTC:2:120]" *.jpg

For each jpg with a non-existent Caption mogrify gives a warning (and continues with the next jpg file).
